net/levin/src/message.rs

```diff
@@ -0,0 +1,198 @@
+//! Levin Messages
+//!
+//! This module contains the [`LevinMessage`], which allows sending bucket body's, full buckets or dummy messages.
+//! The codec will not return [`LevinMessage`] instead it will only return bucket body's. [`LevinMessage`] allows
+//! for more control over what is actually sent over the wire at certain times.
+use bytes::{Bytes, BytesMut};
+
+use crate::{
+    header::{Flags, HEADER_SIZE},
+    Bucket, BucketBuilder, BucketError, BucketHead, LevinBody, LevinCommand, Protocol,
+};
+
+/// A levin message that can be sent to a peer.
+pub enum LevinMessage<T: LevinBody> {
+    /// A message body.
+    /// A levin header will be added to this message before it is sent to the peer.
+    Body(T),
+    /// A full levin bucket.
+    /// This bucket will be sent to the peer directly with no extra information.
+    /// This should only be used to send fragmented messages: [`make_fragmented_messages`]
+    Bucket(Bucket<T::Command>),
+    /// A Dummy message
+    /// A dummy message which the peer will ignore. The dummy message will be the exact size
+    /// (in bytes) of the given `usize` on the wire.
```

Comment:
These won't be spaced out in `cargo docs` btw, it'll be on a single line.

It needs to be like this:
```
/// Line 1
///
/// Line 2
```